### ISSUE TITLE
NPE caused by annotated constructor invocation

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#276]
```
What steps will reproduce the problem?

Use an annotation on a new class expression, that is used as both a parameter 
to a method and has an immediate method call. I was unable to reduce the 
example further.

import checkers.tainting.quals.*;

class Test {
    void test(String s) {
    test(new @Tainted String().toString());
    }
}

What is the expected output? What do you see instead?
Expected no compiler errors.
Received:

An exception has occurred in the compiler (1.8.0-jsr308-1.7.0). Please report 
this bug so we can fix it.  For instructions, see 
http://types.cs.washington.edu/checker-framework/current/README-jsr308.html#repo
rting-bugs .  Thank you.
java.lang.NullPointerException
    at com.sun.tools.javac.comp.Attr.checkForDeclarationAnnotations(Attr.java:2257)
    at com.sun.tools.javac.comp.Attr.visitNewClass(Attr.java:2186)
    at com.sun.tools.javac.tree.JCTree$JCNewClass.accept(JCTree.java:1510)
    at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:598)
    at com.sun.tools.javac.comp.Attr.visitSelect(Attr.java:3244)
    at com.sun.tools.javac.tree.JCTree$JCFieldAccess.accept(JCTree.java:1891)
    at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:598)
    at com.sun.tools.javac.comp.Attr.visitApply(Attr.java:1846)
    at com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1459)
    at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:598)
    at com.sun.tools.javac.comp.DeferredAttr.attribSpeculative(DeferredAttr.java:372)
    at com.sun.tools.javac.comp.DeferredAttr$2.complete(DeferredAttr.java:265)
    at com.sun.tools.javac.comp.DeferredAttr$DeferredType.check(DeferredAttr.java:231)
    at com.sun.tools.javac.comp.DeferredAttr$DeferredType.check(DeferredAttr.java:218)
    at com.sun.tools.javac.comp.Resolve$MethodResultInfo.check(Resolve.java:965)
    at com.sun.tools.javac.comp.Resolve$4.checkArg(Resolve.java:817)
    at com.sun.tools.javac.comp.Resolve$AbstractMethodCheck.argumentsAcceptable(Resolve.java:728)
    at com.sun.tools.javac.comp.Resolve$4.argumentsAcceptable(Resolve.java:826)
    at com.sun.tools.javac.comp.Resolve.rawInstantiate(Resolve.java:573)
    at com.sun.tools.javac.comp.Resolve.selectBest(Resolve.java:1407)
    at com.sun.tools.javac.comp.Resolve.findMethodInScope(Resolve.java:1586)
    at com.sun.tools.javac.comp.Resolve.findMethod(Resolve.java:1657)
    at com.sun.tools.javac.comp.Resolve.findMethod(Resolve.java:1630)
    at com.sun.tools.javac.comp.Resolve.findFun(Resolve.java:1793)
    at com.sun.tools.javac.comp.Resolve$8.doLookup(Resolve.java:2354)
    at com.sun.tools.javac.comp.Resolve$BasicLookupHelper.lookup(Resolve.java:2857)
    at com.sun.tools.javac.comp.Resolve.lookupMethod(Resolve.java:3109)
    at com.sun.tools.javac.comp.Resolve.lookupMethod(Resolve.java:3094)
    at com.sun.tools.javac.comp.Resolve.resolveMethod(Resolve.java:2350)
    at com.sun.tools.javac.comp.Attr.visitIdent(Attr.java:3165)
    at com.sun.tools.javac.tree.JCTree$JCIdent.accept(JCTree.java:2005)
    at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:598)
    at com.sun.tools.javac.comp.Attr.visitApply(Attr.java:1846)
    at com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1459)
    at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:598)
    at com.sun.tools.javac.comp.Attr.attribExpr(Attr.java:647)
    at com.sun.tools.javac.comp.Attr.visitExec(Attr.java:1614)
    at com.sun.tools.javac.tree.JCTree$JCExpressionStatement.accept(JCTree.java:1290)
    at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:598)
    at com.sun.tools.javac.comp.Attr.attribStat(Attr.java:667)
    at com.sun.tools.javac.comp.Attr.attribStats(Attr.java:683)
    at com.sun.tools.javac.comp.Attr.visitBlock(Attr.java:1145)
    at com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:903)
    at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:598)
    at com.sun.tools.javac.comp.Attr.attribStat(Attr.java:667)
    at com.sun.tools.javac.comp.Attr.visitMethodDef(Attr.java:1038)
    at com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:772)
    at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:598)
    at com.sun.tools.javac.comp.Attr.attribStat(Attr.java:667)
    at com.sun.tools.javac.comp.Attr.attribClassBody(Attr.java:4323)
    at com.sun.tools.javac.comp.Attr.attribClass(Attr.java:4233)
    at com.sun.tools.javac.comp.Attr.attribClass(Attr.java:4167)
    at com.sun.tools.javac.comp.Attr.attrib(Attr.java:4142)
    at com.sun.tools.javac.main.JavaCompiler.attribute(JavaCompiler.java:1251)
    at com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:904)
    at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:863)
    at com.sun.tools.javac.main.Main.compile(Main.java:523)
    at com.sun.tools.javac.main.Main.compile(Main.java:381)
    at com.sun.tools.javac.main.Main.compile(Main.java:370)
    at com.sun.tools.javac.main.Main.compile(Main.java:361)
    at com.sun.tools.javac.Main.compile(Main.java:56)
    at com.sun.tools.javac.Main.main(Main.java:42)


What version of the product are you using? On what operating system?
hg changeset:   7223:969c9cf4c550

Linux 3.9.9-0cse.0 #1 SMP Tue Jul 9 14:16:52 PDT 2013 x86_64 x86_64 x86_64 
GNU/Linux

Please provide any additional information below.
In attr
ai.type appears to be the element that is null.
```

Original issue reported on code.google.com by `mcart...@cs.washington.edu` on 18 Nov 2013 at 5:24
